cpp/tests/io/csv_test.cpp

```diff
@@ -227,7 +228,7 @@ void write_csv_helper(std::string const& filename,
                       bool include_header,
                       std::vector<std::string> const& names = {})
 {
-  // write_csv_args is non-owning
+  // csv_writer_options is non-owning
   cudf_io::sink_info const sink{filename};
   std::string const na{"null"};  // why doesn't this have a default?
```

Comment:
This comment is incorrect, the options are owning. You can move this sink_info declaration to the builder call, and completely remove `na` as it is setting the default value. 
This chunk of code here is what actually sparked the idea to improve the API :)

Reply:
but, default value in class is `""` which is derived from the `detail::options`, so we would still need to pass "null".

Reply:
True, my bad. 
However, I replaced "null" with "" here and all test are still passing. I think "null" was used here just because the constructor did not have a default for the `na` parameter. I still think we should keep the default value for `na` here.

Reply:
I updated it with "null" now.

Reply:
Sorry, maybe my previous comment was unclear. I meant to skip setting `na` in this function. The default of "" in `csv_writer_options` is correct - it's consistent with the Python default.